File: ios/App/App/ElearningPlugin.m
//
//  ElearningPlugin.m
//  App
//
//  Created by Michalis Karipidis on 14/5/24.
//

#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(ElearningPlugin, "ElearningScraper", 
    CAP_PLUGIN_METHOD(authenticate, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(apiGet, CAPPluginReturnPromise);
)

File: ios/App/App/ElearningPlugin.swift
import Capacitor

@objc(ElearningPlugin)
public class ElearningPlugin: CAPPlugin {
    @objc func authenticate(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        print("Neron")
        call.resolve(["value": value])
    }
    
    @objc func apiGet(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        print("Michalis!!")
        call.resolve(["value": value])
    }
}
